hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/translators/BlockTransactionalUnitTranslator.java

```diff
@@ -0,0 +1,164 @@
+        translators.put(HederaFunctionality.TOKEN_FEE_SCHEDULE_UPDATE, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_REJECT, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_ACCOUNT_WIPE, new TokenWipeTranslator());
+        translators.put(HederaFunctionality.TOKEN_ASSOCIATE_TO_ACCOUNT, new TokenAssociateTranslator());
+        translators.put(HederaFunctionality.TOKEN_DISSOCIATE_FROM_ACCOUNT, new TokenDissociateTranslator());
+        translators.put(HederaFunctionality.TOKEN_GRANT_KYC_TO_ACCOUNT, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_REVOKE_KYC_FROM_ACCOUNT, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_BURN, new TokenBurnTranslator());
+        translators.put(HederaFunctionality.TOKEN_CREATE, new TokenCreateTranslator());
+        translators.put(HederaFunctionality.TOKEN_DELETE, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_FREEZE_ACCOUNT, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_UNFREEZE_ACCOUNT, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_UPDATE_NFTS, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_MINT, new TokenMintTranslator());
+        translators.put(HederaFunctionality.TOKEN_PAUSE, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_UNPAUSE, NO_EXPLICIT_SIDE_EFFECTS_TRANSLATOR);
+        translators.put(HederaFunctionality.TOKEN_UPDATE, new TokenUpdateTranslator());
+        translators.put(HederaFunctionality.UTIL_PRNG, new UtilPrngTranslator());
+    }
+
+    /**
+     * Scans a block for genesis information and returns true if found.
+     * @param block the block to scan
+     * @return true if genesis information was found
+     */
+    public boolean scanBlockForGenesis(@NonNull final Block block) {
+        return baseTranslator.scanMaybeGenesisBlock(block);
+    }
+
+    /**
+     * Translates the given {@link BlockTransactionalUnit} into a list of {@link SingleTransactionRecord}s.
+     * @param unit the unit to translate
+     * @return the translated records
+     */
+    public List<SingleTransactionRecord> translate(@NonNull final BlockTransactionalUnit unit) {
+        requireNonNull(unit);
+        baseTranslator.prepareForUnit(unit);
+        final List<StateChange> remainingStateChanges = new LinkedList<>(unit.stateChanges());
+        final List<SingleTransactionRecord> translatedRecords = new ArrayList<>();
+        for (final var blockTransactionParts : unit.blockTransactionParts()) {
+            final var translator = translators.get(blockTransactionParts.functionality());
+            if (translator == null) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.warn(
```

Comment:
Perhaps we should throw an exception here, if no translator is known for the `HederaFunctionality` then we would be unable to add the approriate `TransactionRecord`'s to the cache.